src/System.Drawing.Common/src/System/Drawing/BufferedGraphicsContext.cs

```diff
@@ -0,0 +1,147 @@
+
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+
+namespace System.Drawing
+{
+    /// <summary>
+    /// The BufferedGraphicsContext class can be used to perform standard double buffer rendering techniques.
+    /// </summary>
+    public sealed partial class BufferedGraphicsContext : IDisposable
+    {
+        private Size _maximumBuffer;
+        private Size _bufferSize = Size.Empty;
+
+        private static TraceSwitch s_doubleBuffering;
+
+        /// <summary>
+        /// Basic constructor.
+        /// </summary>
+        public BufferedGraphicsContext()
+        {
+            // By defualt, the size of our maxbuffer will be 3 x standard button size.
+            _maximumBuffer.Width = 75 * 3;
+            _maximumBuffer.Height = 32 * 3;
+        }
+
+        /// <summary>
+        /// Internal trace switch for debugging
+        /// </summary>
+        internal static TraceSwitch DoubleBuffering
+        {
+            get
+            {
+                if (s_doubleBuffering == null)
+                {
+                    s_doubleBuffering = new TraceSwitch("DoubleBuffering", "Output information about double buffering");
+                }
+
+                return s_doubleBuffering;
+            }
+        }
+
+        /// <summary>
+        /// Allows you to set the maximum width and height of the buffer that will be retained in memory.
+        /// You can allocate a buffer of any size, however any request for a buffer that would have a total
+        /// memory footprint larger that the maximum size will be allocated temporarily and then discarded 
+        /// with the BufferedGraphics is released.
+        /// </summary>
+        public Size MaximumBuffer
+        {
+            get => _maximumBuffer;
+            set
+            {
+                if (value.Width <= 0 || value.Height <= 0)
+                {
+                    throw new ArgumentException(SR.Format(SR.InvalidArgumentValue, nameof(MaximumBuffer), value), nameof(value));
+                }
+
+                // If we've been asked to decrease the size of the maximum buffer,
+                // then invalidate the older & larger buffer.
+                if (value.Width * value.Height < _maximumBuffer.Width * _maximumBuffer.Height)
+                {
+                    Invalidate();
+                }
+
+                _maximumBuffer = value;
+            }
+        }
+
+        ~BufferedGraphicsContext() => Dispose(false);
+
+        /// <summary>
+        /// Returns a BufferedGraphics that is matched for the specified target Graphics object.
+        /// </summary>
+        public BufferedGraphics Allocate(Graphics targetGraphics, Rectangle targetRectangle)
+        {
+            if (ShouldUseTempManager(targetRectangle))
+            {
+                Debug.WriteLineIf(DoubleBuffering.TraceWarning, "Too big of buffer requested (" + targetRectangle.Width + " x " + targetRectangle.Height + ") ... allocating temp buffer manager");
```

Comment:
Should we remove these? Looks like legacy code...

Reply:
If we remove this then `DoubleBuffering` would be Windows specific property. Feel free to remove them if you're certain it is legacy code. I don't really know if it is.